src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -111,20 +111,20 @@ internal Task InitializeJobhostLanguageWorkerChannelAsync(int attemptCount)
             var languageWorkerChannel = _languageWorkerChannelFactory.CreateLanguageWorkerChannel(_scriptOptions.RootScriptPath, _workerRuntime, _metricsLogger, attemptCount, _managedDependencyOptions);
             languageWorkerChannel.SetupFunctionInvocationBuffers(_functions);
             _jobHostLanguageWorkerChannelManager.AddChannel(languageWorkerChannel);
-            languageWorkerChannel.StartWorkerProcessAsync()
-                 .ContinueWith(workerInitTask =>
-                 {
-                     if (workerInitTask.IsCompleted)
-                     {
-                         _logger.LogDebug("Adding jobhost language worker channel for runtime: {language}. workerId:{id}", _workerRuntime, languageWorkerChannel.Id);
-                         languageWorkerChannel.SendFunctionLoadRequests();
-                         State = FunctionDispatcherState.Initialized;
-                     }
-                     else
-                     {
-                         _logger.LogWarning("Failed to start language worker process for runtime: {language}. workerId:{id}", _workerRuntime, languageWorkerChannel.Id);
-                     }
-                 });
+            _logger.LogDebug("Adding jobhost language worker channel for runtime: {language}. workerId:{id}", _workerRuntime, languageWorkerChannel.Id);
+            // Not awaiting
+            languageWorkerChannel.StartWorkerProcessAsync();
+            languageWorkerChannel.SendFunctionLoadRequests().ContinueWith(functionLoadTask =>
+                {
+                    if (functionLoadTask.IsCompleted)
+                    {
+                        State = FunctionDispatcherState.Initialized;
+                    }
+                    else
+                    {
+                        _logger.LogWarning("Failed to start language worker process for runtime: {language}. workerId:{id}", _workerRuntime, languageWorkerChannel.Id);
+                    }
+                });
```

Comment:
I am not sure if this is the right approach. We aren't awaiting, which I think is intentional?

Reply:
function invocations are buffered until function load responses are received from the worker.